bridge-history-api/internal/logic/l1_fetcher.go

```diff
@@ -45,11 +45,14 @@ type L1FetcherLogic struct {
 	crossMessageOrm *orm.CrossMessage
 	batchEventOrm   *orm.BatchEvent
 
+	lastFinalizedBatchIndex            *uint64
+	nextPendingLastFinalizedBatchIndex uint64
```

Comment:
we can remove them if using `batch_index <= x && batch_status != finalized` to update db.

Reply:
finalization can be treated as deterministic.